survival/utils.py

```diff
@@ -0,0 +1,31 @@
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+def intersect_series_with_index(index: pd.Index, data: pd.Series) -> pd.Series:
+    """
+    returns series intersected with given index
+    """
+    aligned_series = data.loc[index.intersection(data.index)]
+    return aligned_series
+
+def intersect_df(dataframes=()):
+    """
+    returns all dataframe/series-s with intersected indexes in the same order
+    :param dataframes: list of pd.DataFrame/pd.Series
+    """
+    shared_indices = set(dataframes[0].index)
+    for df_index in range(1, len(dataframes)):
```

Comment:
no need to iterate over in index:
`for df in dataframes[1:]:`

also change inside the loop